### PR TITLE
vscode-extensions.2gua.rainbow-brackets: init at 0.0.6

### PR DESCRIPTION
**Pull request by CompEng0001**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by CompEng0001** [comment]
bad PR on my part.

**Comment by CompEng0001** [comment]
Closed: [https://github.com/NixOS/nixpkgs/pull/185684]( https://github.com/NixOS/nixpkgs/pull/185684) 

New PR from different branch.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -47,6 +47,22 @@ let
         };
       };
 
+      _2gua.rainbow-brackets = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          publisher = "2gua"
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:52]:
```suggestion
          publisher = "2gua";
```

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 2]
```diff
@@ -47,6 +47,22 @@ let
...
+          name = "Rainbow Brackets";
+          version = "0.0.6";
+          sha256 = "3svGVbaxUh/60NywMdWFsL3fSv+kPgt39bGmPVIu6sI="
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:55]:
```suggestion
          sha256 = "3svGVbaxUh/60NywMdWFsL3fSv+kPgt39bGmPVIu6sI=";
```

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 3]
```diff
@@ -47,6 +47,22 @@ let
...
+        };
+        meta = {
+          description = "Provide rainbow colors for the round brackets, the square brackets and the squiggly brackets. This is particularly useful for Lisp or Clojure programmers, and of course, JavaScript, and other programmer";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:58]:
```suggestion
          description = "A Visual Studio Code extension providing rainbow brackets";
```

**Comment by superherointj** [comment]
[mod: added CompEng0001 as maintainer for package](https://github.com/NixOS/nixpkgs/pull/189331/commits/cc1c12e1490cc879be9c4a771e329530916dbb5c) is empty. And title is not suitable.

Suggestion for commit: `maintainers: add CompEng0001`

**Comment by superherointj** [comment]
Certify to build package before you submit a PR for review. I have suggested fixes here but still, you have to re-build package after every change to be sure your changes have not created problems. And always squash commits, ideally keep one commit per package.

**Comment by superherointj** [comment]
Please, squash commits. (You've got two messed commits.)

Then, add yourself to maintainers list. The file is:
https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix

Use as commit title:
> `maintainers: add CompEng0001`

**Comment by CompEng0001** [comment]
> Please, squash commits. (You've got two commits editing same file.)
> 
> Then, add yourself to maintainers list. The file is: https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix
> 
> The second commit title (that adds you as maintainer) should be:
> 
> > `maintainers: add CompEng0001`
Second PR -> https://github.com/NixOS/nixpkgs/pull/189992

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 4]
```diff
@@ -47,6 +47,22 @@ let
         };
       };
 
+      _2gua.rainbow-brackets = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          publisher = "2gua";
+          name = "Rainbow Brackets";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:53]: `name` should have been `rainbow-brackets`.

**Comment by superherointj** [comment]
> Second PR -> #189992

Please, refrain from opening new PRs. It loses review history. And makes reviewer life hell.

Also, build the package. Again I found mistakes that if you were building the package yourself this would not be happening.

To build:
* `nix build .#vscode-extensions.2gua.rainbow-brackets` (flakes command)

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 5]
```diff
@@ -47,6 +47,22 @@ let
...
+          version = "0.0.6";
+          sha256 = "3svGVbaxUh/60NywMdWFsL3fSv+kPgt39bGmPVIu6sI=";
+        };
+        meta = {
```
[the remark this answers is not in the document]
Comment (CompEng0001) [reply pkgs/applications/editors/vscode/extensions/default.nix]: Done.

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 4 above)
Comment (CompEng0001) [reply pkgs/applications/editors/vscode/extensions/default.nix]: done

**Comment by CompEng0001** [comment]
OK yeah...

So I ran 

 - `vscode-extensions._2gua.rainbow-brackets: init at 0.0.6` and this runs fine now that I have correctedthere errors.
 
 Is the naming convention right, because _1Password.op-vscode only builds when the  '_' is used?

**Comment by superherointj** [comment]
> Is the naming convention right, because _1Password.op-vscode only builds when the '_' is used?

Cannot start with number.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 189331` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vscode-extensions._2gua.rainbow-brackets</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]

**Pull request merged**